test/driver.js

```diff
@@ -582,6 +584,18 @@ var Driver = (function DriverClosure() { // eslint-disable-line no-unused-vars
       this._log('Done !');
       this.end.textContent = 'Tests finished. Close this window!';
 
+      var re = new XMLHttpRequest();
+      re.open('POST', '/browserTestReports', false);
+      re.setRequestHeader('Content-Type', 'application/json');
+      re.responseType = 'arraybuffer';
+      re.onreadystatechange = function () {
```

Comment:
Once again, I strongly suggest to read the documentation to understand what this API does. Your current implementation is not sending any data because the `re.send` call happens inside a `onreadystatechange` event handler, but this `onreadystatechange` event handler is never fired because the request is never started by a [`.send`](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/send) call.

If code is not working as expected, take a step back and try to understand why your code is not working as expected. That's likely a better learning experience than copying code from examples without actually understanding the purpose of the copied code. Initially, this will take more time, but in the long term this strategy will make you more knowledgeable and enables you to work independently.

Reply:
after reading all the suggested docs I've made a few minor changes.